src/cells/model.ts

```diff
@@ -182,6 +192,26 @@ class BaseCellModel implements IBaseCellModel {
   }
 
   /**
+   * Get whether the cell is focused for editing.
+   *
+   * #### Notes
+   * This is a delegate to the focused state of the input's editor.
+   */
+  get focused(): boolean {
+    return this.input.textEditor.focused;
+  }
+
+  /**
+   * Get whether the cell is focused for editing.
```

Comment:
Actually, I prefer `mode: 'command' | 'edit'` to correspond with the notebook level attribute.